ingestion/src/metadata/clients/domo_client.py

```diff
@@ -103,14 +103,10 @@ def __init__(
         ],
     ):
         self.config = config
-        self.config.instanceDomain = (
-            self.config.instanceDomain[:-1]
-            if self.config.instanceDomain.endswith("/")
-            else self.config.instanceDomain
-        )
         HEADERS.update({"X-DOMO-Developer-Token": self.config.accessToken})
         client_config: ClientConfig = ClientConfig(
-            base_url=self.config.instanceDomain,
+            # AnyUrl string ends with / and the domo API does not respond properly if it has 2 // at the end
+            base_url=str(self.config.instanceDomain)[:-1],
```

Comment:
IMO the previous code was correct here 🤔 in case the user voluntarily or by mistakenly puts `/` in the end then we are trying to remove that but if user does not put the `/` end we will be modifying an actual url right

for example `http://localhost:8585/` ->  `http://localhost:8585`
but `http://localhost:8585`  -> `http://localhost:858`

we can use this function here
https://github.com/open-metadata/OpenMetadata/blob/e9ec4367ee42ea00c1345505c22d7d2c7247ba9f/ingestion/src/metadata/utils/helpers.py#L354

Reply:
I added a "test" about this here https://github.com/open-metadata/OpenMetadata/pull/16480/files#diff-ea5a94c37687536bba98a30863374a14b14107ba16069c681e8610eba3c72521R87 (pasting in case the files load slow)

```python
def test_any_url():
    """It always ends with /"""
    assert str(AnyUrl("https://example.com")) == "https://example.com/"
    assert str(AnyUrl("https://example.com/")) == "https://example.com/"
```

Reply:
ohhh nice